cpp/ql/test/library-tests/dataflow/fields/D.cpp

```diff
@@ -0,0 +1,66 @@
+void sink(void *o);
+
+class D {
+public:
+  struct Elem { };
+
+  struct Box1 {
+    Elem *elem;
+    Box1(Elem *e) { elem = e; }
+    Elem* getElem() { return elem; }
+    void setElem(Elem *e) { elem = e; }
+  };
+
+  struct Box2 {
+    Box1* box;
+    Box2(Box1* b) { box = b; }
+    Box1* getBox1() { return box; }
+    void setBox1(Box1* b) { box = b; }
+  };
+
+  static void sinkWrap(Box2* b2) {
+    sink(b2->getBox1()->getElem());
+  }
+
+  Box2* boxfield;
+
+  void f1() {
+    Elem* e = new Elem(); // source of flow to sinkWrap
+    Box2* b = new Box2(new Box1(nullptr));
```

Comment:
It's a little unfortunate that the test considers all calls to `new`, including this one, to be a source.

Reply:
I completely agree. To interpret the test results it's necessary to look at the `edges` relation and see that flow passes through the intended fields. This is a problem inherited from the Java tests, and it also affects `[ABC].cpp`.

If we want to address it, we should do it for all those files and possibly even for Java and C#. I don't think this PR is the place to fix it.

Reply:
Agreed.  Merging...

Reply:
Ah, there's a conflict.